### ISSUE TITLE
Date Filter is passing in +05:30 hr fomat instead normal date

### ISSUE THREAD

**Issue by Ravi9535** [metabase/metabase#40775]
### Describe the bug

When user select date in the metabase filter it should pass as normal date like if they pass 29-03-2025 it should fetch records that satisfy that date But Metabase passing it as "2024-03-27 18:30:00+05:30" when date is compared it is showing 1 day before data
![date](https://github.com/metabase/metabase/assets/66297271/d514313b-f9ed-4b26-a469-23b49e0d810e)

### To Reproduce

1. Go to 'metabase SQL query'
2. Click on 'write query with some date make sure that date has some unwated prefix'
3. See error

### Expected behavior

It should pass date as what date we pass in filter 

### Information about your Metabase installation

```JSON
- Metabase 0.49.2
- Chome Version 122.0.6261.111 (Official Build) (64-bit)
```

### Severity

User want report now itself

**Comment by paoliniluis** [comment]
What's your Metabase reporting timezone?

**Comment by Ravi9535** [comment]
Actually we use 2 database and the time zone in metabase is **Asia/Culcutta**
1.Mysql 
here in DB we are using +05:30 hr format
2.Amazon redsshift
Here we are storing the the date as it is not in +05:30 Hr format

So when I select 2024-03-29 it will check for 2024-03-28 18:30:00+05:30

**Comment by ranquild** [comment]
The FE sends the correct parameter value (just the date).

**Comment by paoliniluis** [comment]
I "think" that the problem is that the error returned is wrong. The issue here is that Redshift does not seem to like a date + an integer
![image](https://github.com/metabase/metabase/assets/1711649/e5ab2bb3-bdcc-4601-a2cd-ed6cc8deb955)

**Comment by camsaul** [comment]
So believe it or not,

```sql
SELECT date '2024-07-02' + 1
```

actually does work in Redshift (and Postgres), it seems to be sorta like

```sql
SELECT date '2024-07-02' + interval '1 day'
```

altho `date + int` returns a `date` while `date + interval` returns a `timestamp`.

This was fixed in 0.49.8 in #41864.
